<?xml version="1.0"?>
<launch>
    <node name="rplidar_node" pkg="rplidar_ros" type="rplidarNode" output="screen">
        <param name="serial_port" type="string" value="/dev/ttyUSB0"/>
        <param name="serial_baudrate" type="int" value="115200"/>
        <param name="frame_id" type="string" value="laser_frame"/>
        <param name="inverted" type="bool" value="false"/>
        <param name="angle_compensate" type="bool" value="true"/>
    </node>

    <param name="robot_description" textfile="$(find my_robot)/urdf/robot.urdf" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="50.0"/>
    </node>

    <rosparam file="$(find my_robot)/config/ros_control.yaml" command="load"/>

    <node name="pi_control_node"
          pkg="my_robot_control"
          type="pi_control_node"
          output="screen"/>

    <node name="cmd_vel_translator"
          pkg="my_robot"
          type="cmd_vel_translator.py"
          output="screen"/>

    <node name="controller_spawner"
          pkg="controller_manager"
          type="spawner"
          respawn="false"
          output="screen"
          args="joint_state_controller
                left_wheel_velocity_controller
                right_wheel_velocity_controller"/>

    <node name="odom_publisher"
          pkg="my_robot"
          type="odom_publisher.py"
          output="screen"/>
</launch>
